### PR TITLE
Migrated ExportImportTest and associated files

### LINKED ISSUE #48152: Migrate the `ExportImportTest`
### Description

Migrate the `org.keycloak.tests.exportimport.ExportImportTest` test

### PR DESCRIPTION
**Pull request by msdaly200**
Migrated ExportImportTest and associated files

Closes #48152

### CODE REVIEW

**Comment by vaceksimon** [comment]
Please rebase and resolve conflicts

File: tests/base/src/test/java/org/keycloak/tests/authz/admin/permissions/ExportImportTestFGAPV2.java [hunk 1]
```diff
@@ -33,14 +40,21 @@
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
-public class ExportImportTestFGAPV2 extends ExportImportTest {
+@KeycloakIntegrationTest(config = ExportImportTestFGAPV2.KeycloakAdminPermissionsV2ServerConfig.class)
+public class ExportImportTestFGAPV2 {
```
Comment (rmartinc) [inline tests/base/src/test/java/org/keycloak/tests/authz/admin/permissions/ExportImportTestFGAPV2.java]: I don't see this test executed in the new jobs. Probably you need to include the package in the `Base2TestSuite`.

File: tests/custom-providers/src/main/resources/META-INF/services/org.keycloak.events.EventListenerProviderFactory [hunk 2]
```diff
@@ -1 +1,2 @@
-org.keycloak.tests.providers.events.TestEventsListenerContextDetailsProviderFactory
\ No newline at end of file
+org.keycloak.tests.providers.events.TestEventsListenerContextDetailsProviderFactory
+org.keycloak.tests.providers.events.TestEventsListenerProviderFactory
```
Comment (rmartinc) [inline tests/custom-providers/src/main/resources/META-INF/services/org.keycloak.events.EventListenerProviderFactory]: Do you really need to create this factory? Try to remove the factory from this file and also remove the two classes (TestEventsListenerProvider and TestEventsListenerProviderFactory). You can also remove any reference in the json files:

```diff
-  "eventsListeners": ["jboss-logging", "event-queue"]
+  "eventsListeners": ["jboss-logging"]
```

Check if that works and we can remove the events classes from this PR.

File: tests/base/src/test/java/org/keycloak/tests/authz/admin/permissions/ExportImportTestFGAPV2.java [hunk 3]
```diff
@@ -33,14 +40,21 @@
...
+
+    @InjectRealm
+    ManagedRealm managedRealm;
```
Comment (rmartinc) [inline tests/base/src/test/java/org/keycloak/tests/authz/admin/permissions/ExportImportTestFGAPV2.java]: Maybe you can remove this managed realm, I think that the tests do not use any realm at all and can be deleted.

**Review by rmartinc — CHANGES_REQUESTED** [review]
Thanks a lot @msdaly200! Just a few suggestions.

File: tests/custom-providers/src/main/resources/META-INF/services/org.keycloak.events.EventListenerProviderFactory (hunk 2 above)
Comment (msdaly200) [reply tests/custom-providers/src/main/resources/META-INF/services/org.keycloak.events.EventListenerProviderFactory]: Hi @rmartinc so that's fine, I removed the reference from the migrated tests/base/src/test/resources/testrealm.json so no need for the files you mention above, have removed those.

File: tests/base/src/test/java/org/keycloak/tests/authz/admin/permissions/ExportImportTestFGAPV2.java (hunk 1 above)
Comment (msdaly200) [reply tests/base/src/test/java/org/keycloak/tests/authz/admin/permissions/ExportImportTestFGAPV2.java]: ExportImportTestFGAPV2 should be in the Base2TestSuite as part of this package already in the file, "org.keycloak.tests.authz", - I'll double check with the next run

File: tests/base/src/test/java/org/keycloak/tests/authz/admin/permissions/ExportImportTestFGAPV2.java (hunk 3 above)
Comment (msdaly200) [reply tests/base/src/test/java/org/keycloak/tests/authz/admin/permissions/ExportImportTestFGAPV2.java]: Thanks @rmartinc for the review

File: tests/base/src/test/java/org/keycloak/tests/authz/admin/permissions/ExportImportTestFGAPV2.java (hunk 1 above)
Comment (msdaly200) [reply tests/base/src/test/java/org/keycloak/tests/authz/admin/permissions/ExportImportTestFGAPV2.java]: Hi @rmartinc, so you were right, renaming the test to ExportImportFGAPV2Test (i.e. end in test) does make it get picked up, so can now see it in the new [base2testsuite](https://github.com/keycloak/keycloak/actions/runs/26571610431/job/78280762033?pr=49338)  and tests are passing.

**Review by rmartinc — DISMISSED** [review]
Thanks @msdaly200! LGTM! @vaceksimon please check the PR when you have time.

File: tests/base/src/test/java/org/keycloak/tests/exportimport/ExportImportTest.java [hunk 4]
```diff
@@ -439,7 +375,11 @@ public void testImportIgnoreExistingMissingClientId() {
         }
     }
 
-    @UncaughtServerErrorExpected
```
Comment (vaceksimon) [inline tests/base/src/test/java/org/keycloak/tests/exportimport/ExportImportTest.java:442]: That's funny. This method doesn't have the `@Test` annotation so it was never executed.
I think we can fix this test by just adding an assertThrows for the last statement as was supposed to be done in https://github.com/keycloak/keycloak/commit/827e82ad2574b97df47d119fe551d452cba3b36e

**Comment by msdaly200** [comment]
Hi @vaceksimon could you review? Thanks!

**Review by vaceksimon — CHANGES_REQUESTED** [review]

File: tests/base/src/test/java/org/keycloak/tests/exportimport/ExportImportTest.java (hunk 4 above)
Comment (rmartinc) [reply tests/base/src/test/java/org/keycloak/tests/exportimport/ExportImportTest.java:442]: :smile: Yes, you can enable the test again and change the final line with something like:

```java
            RunOnServerException e = Assertions.assertThrows(RunOnServerException.class, () -> runOnServerMaster.run(ExportImportHelper.runImport()));
            assertThat(e.getMessage(), Matchers.containsString("File name / realm name mismatch."));
```

File: tests/base/src/test/java/org/keycloak/tests/exportimport/ExportImportTest.java [hunk 5]
```diff
@@ -439,7 +376,9 @@ public void testImportIgnoreExistingMissingClientId() {
...
+    // The test should verify that import works when filename doesn't match realm name
+    // Revisit when the server-side functionality is fixed
```
Comment (vaceksimon) [inline tests/base/src/test/java/org/keycloak/tests/exportimport/ExportImportTest.java]: Correct me if I'm wrong, but the commit I previously linked suggests that this test actually verifies that import **doesn't work** when the filename doesn't match the realm name. Which means this is exactly as intended and there is no need to fix the server-side functionality.

File: tests/base/src/test/java/org/keycloak/tests/exportimport/ExportImportTest.java (hunk 4 above)
Comment (msdaly200) [reply tests/base/src/test/java/org/keycloak/tests/exportimport/ExportImportTest.java:442]: I'd commited the change before I saw your comment @rmartinc, so there is now one more commit :-)

File: tests/base/src/test/java/org/keycloak/tests/exportimport/ExportImportTest.java (hunk 5 above)
Comment (rmartinc) [inline tests/base/src/test/java/org/keycloak/tests/exportimport/ExportImportTest.java]:
```suggestion
```

File: tests/base/src/test/java/org/keycloak/tests/exportimport/ExportImportTest.java [hunk 6]
```diff
@@ -439,7 +376,9 @@ public void testImportIgnoreExistingMissingClientId() {
...
+    @Test
     public void testImportNameMismatch() {
         runOnServerMaster.run(ExportImportHelper.setStrategy(Strategy.IGNORE_EXISTING));
         runOnServerMaster.run(ExportImportHelper.setProvider(DirExportProviderFactory.PROVIDER_ID));
```
Comment (rmartinc) [inline tests/base/src/test/java/org/keycloak/tests/exportimport/ExportImportTest.java:382]:
```suggestion
        runOnServerMaster.run(ExportImportHelper.setProvider(DirExportProviderFactory.PROVIDER_ID));
        runOnServerMaster.run(ExportImportHelper.setRealmName(null));
```

**Review by rmartinc — CHANGES_REQUESTED** [review]
I think those lines are wrong...

**Comment by rmartinc** [comment]
@msdaly200 I have applied the changes, let's see the tests...

**Review by rmartinc — APPROVED** [review]
Nice! It worked!

File: tests/base/src/test/java/org/keycloak/tests/exportimport/ExportImportTest.java (hunk 5 above)
Comment (rmartinc) [reply tests/base/src/test/java/org/keycloak/tests/exportimport/ExportImportTest.java]: I used github to remove the lines and add a nother one to make the test pass (it seems another test is setting the realm and we need to set it to null).

**Review by vaceksimon — APPROVED** [review]
